### PR TITLE
Query Matomo through its API and insert results into the portal's database tables

### PR DESCRIPTION
**Pull request by cmadjar**
### Description

Analytics are gathered by our instance of Matomo. This PR adds new tables to `models.py` used by flask to create the database tables that will store the Matomo API responses identified in #405, #404 and #406. The information stored in those tables will be used to plot the following analytics graphs:
- number of visits/uniq visitors per week, months over time
- number of views per dataset 
- number of views per pipeline
- popular site search keywords

### How to populate the tables

- 1) create the tables: `flask db upgrade`
- 2) populate the table content by querying Matomo: `flask update_analytics`

##### 1)  `matomo_daily_visits_summary` 
This table will store all the stats reported by Matomo regarding daily visits to the portal. Columns of the table are:
```
    id                   = ID
    date                 = date 
    avg_time_on_site     = average time spent, in seconds, on this page
    bounce_count         = number of visits that bounced (viewed only one page)
    max_actions          = maximum number of actions in a visit
    nb_actions           = number of actions (page views, outlinks and downloads)
    nb_uniq_visitors     = number of unique visitors
    nb_users             = number of unique active users (visitors with a known User ID).
                           If you are not using User ID then this metric will be set to zero.
    nb_visits            = number of visits (30 min of inactivity considered a new visit)
    nb_visits_converted  = number of visits that converted a goal
    sum_visit_length     = total time spent, in seconds
```

##### 2) `matomo_daily_dataset_page_views_summary`
This table will store all the daily stats for each dataset_id. Columns are:
```
    id               = ID
    date             = date
    dataset_id       = dataset_id as stored in datasets table
    url              = page URL
    label            = page label
    nb_hits          = number of views on this page
    nb_visits        = number of visits (30 min of inactivity considered a new visit)
    nb_uniq_visitors = number of unique visitors
    sum_time_spent   = total time spent on this page, in seconds
    avg_time_on_page = average time spent, in seconds, on this page
```

##### 3)  `matomo_daily_get_page_urls_summary`
This table will store all the daily stats for all URLs except for the dataset pages. Columns are: 
```
    id               = ID
    date             = date
    url              = page URL
    label            = page label
    nb_hits          = number of views on this page
    nb_visits        = number of visits (30 min of inactivity considered a new visit)
    nb_uniq_visitors = number of unique visitors
    sum_time_spent   = total time spent on this page, in seconds
    avg_time_on_page = average time spent, in seconds, on this page
```

##### 4) `matomo_daily_site_keyword_searches_summary`
This table will store all the daily site search keyword stats. Columns are:
```
    id                  = ID
    date                = date
    avg_time_on_page    = average time spent, in seconds, on this page
    bounce_rate         = ratio of visits leaving the website after landing on this page
    exit_nb_visits      = number of visits that finished on this page
    exit_rate           = ratio of visits that do not view any other page after this page
    label               = keyword searched
    nb_hits             = number of views on this page
    nb_pages_per_search = number of pages displayed for the searched keyword
    nb_visits           = number of visits (30 min of inactivity considered a new visit)
    segment             = segment with keyword search
    sum_time_spent      = total time spent on this page, in seconds
```

## Related issues

#404, #405, #406

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -67,7 +67,8 @@ class Config(object):
 
     # Matomo
     MATOMO_SERVER_URL = os.environ.get("MATOMO_SERVER_URL")
-    MATOMO_SITE_ID = os.environ.get("MATOMO_SITE_ID", "2")
+    MATOMO_SITE_ID    = os.environ.get("MATOMO_SITE_ID", "2")
+    MATOMO_TOKEN_AUTH = os.environ.get("MATOMO_TOKEN_AUTH")
```
Comment (cmadjar) [inline config.py:71]: @liamocn this is the config to get the token auth of matomo.

**Comment by cmadjar** [comment]
@liamocn ready for review

**Comment by cmadjar** [comment]
@liamocn rebased and passing circle CI. All ready for you. :)

File: app/cli.py [hunk 2]
```diff
@@ -305,3 +312,286 @@ def _update_datasets(app):
...
+            keyword_summary.nb_hits = keyword['nb_hits']
+            keyword_summary.nb_pages_per_search = keyword['nb_pages_per_search']
+            keyword_summary.nb_visits = keyword['nb_visits']
+            keyword_summary.segment = keyword['segment']
+            keyword_summary.sum_time_spent = keyword['sum_time_spent']
+
+            db.session.merge(keyword_summary)
+            db.session.commit()
+
+    print(f'[INFO   ] Inserted Matomo search keywords summary for {date}')
```
Comment (liamocn) [inline app/cli.py:576]: The indentation here is wrong, this belongs to the for block above.

**Review by liamocn — COMMENTED** [review]
Just one quick change. Otherwise, looks good!

File: app/cli.py (hunk 2 above)
Comment (cmadjar) [reply app/cli.py]: Good catch @liamocn. Thanks! Fixed in the last commit.

**Pull request merged**